executor/hnswlib_searcher.py

```diff
@@ -347,3 +352,34 @@ def sync(self, delta: GENERATOR_DELTA):
                 self.update(da)
 
         self.last_timestamp = datetime.now()
+
+    def index_sync(self, iterator: GENERATOR_DELTA, batch_size=100) -> None:
+        if iterator is None:
+            self.logger.warning('No data received in HNSW.sync. Skipping...')
+            return
+
+        this_batch_size = 0
+        # batching
+        this_batch_embeds = np.zeros((batch_size, self.dim), dtype=HNSW_TYPE)
+        this_batch_ids = []
+
+        while True:
+            try:
+                doc_id, vec_array, _ = next(iterator)
+                vec = vec_array.astype(HNSW_TYPE)
```

Comment:
We need filter out the soft-deleted documents
```suggestion
                if vec_array is None:
                    continue
                vec = vec_array.astype(HNSW_TYPE)
```